cohttp-eio/src/cohttp_eio.mli

```diff
@@ -112,20 +112,27 @@ end
 module Client : sig
   type response = Http.Response.t * Eio.Buf_read.t
 
-  type host = string * int option
-  (** Represents a server host - as ip address or domain name - and an optional
-      port value, e.g. www.example.org:8080, www.reddit.com *)
+  type host = string
+  (** Represents a server host - as ip address or domain name, e.g.
+      www.example.org:8080, www.reddit.com*)
+
+  type port = int
+  (** Represents a tcp/ip port value *)
 
   type resource_path = string
   (** Represents HTTP request resource path, e.g. "/shop/purchase",
       "/shop/items", "/shop/categories/" etc. *)
 
-  type 'a body_disallowed_call =
+  type 'a env = < net : Eio.Net.t ; .. > as 'a
+
+  type ('a, 'b) body_disallowed_call =
     ?pipeline_requests:bool ->
     ?version:Http.Version.t ->
     ?headers:Http.Header.t ->
-    conn:(#Eio.Flow.two_way as 'a) ->
-    host ->
+    ?conn:(#Eio.Flow.two_way as 'a) ->
+    ?port:port ->
+    'b env ->
+    host:host ->
```

Comment:
why did we decide to not use `uri` here?
A `uri` will contain all the information at once

Reply:
Not really sure now but if I remember right being able to use the `flow/connection` was one I think. It wasn't obvious to me then but now I think it was the correct design. Additionally, I believe we can't use `unix` domain sockets in url without inventing our own syntax since http uri spec.  doesn't specify how we can use `unix` domain socket scheme in the uri.

Reply:
That makes sense, the main reason I am asking is because right now it feels weird to separate the host from the path (mostly because any other library will take that all at once) but this is surely an improvement over the previous version

Reply:
Should we try to be smart and add `/` as the first character if not already present?

Reply:
> Should we try to be smart and add / as the first character if not already present?

I think that is a good idea but not sure atm. However, I would like to address that in future PR please if we do go down that route.

Reply:
I have created a new issue so that we can discuss it further. https://github.com/mirage/ocaml-cohttp/issues/959